metasrv/src/metrics/meta_metrics.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Once;
+
+use common_metrics::Gauge;
+use common_metrics::MetricOption;
+use once_cell::sync::OnceCell;
+
+pub const META_NAMESPACE: &str = "metasrv";
+pub const SERVER_SUBSYSTEM: &str = "server";
+
+pub struct MetaMetrics {
+    has_leader: Option<Gauge>,
+}
+
+static INSTANCE: OnceCell<MetaMetrics> = OnceCell::new();
+
+impl Default for MetaMetrics {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```

Comment:
I think this can be derived.

But it's not a big deal and should not be a blocker for this PR.